src/Diagnostics.RuntimeHost/SerializationConvertors/AsRuntimeTypeConverter.cs

```diff
@@ -42,22 +42,23 @@ public override void Write(Utf8JsonWriter writer, T value, JsonSerializerOptions
         }
     }
 
-    public class AsPrintableTypeConverter<T> : JsonConverter<T>
+    public class ExceptionConverter : JsonConverter<Exception>
     {
-        private static JsonSerializerOptions _serializeWithoutFieldOptions = new JsonSerializerOptions
+        public override Exception Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
         {
-            PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
-            WriteIndented = true
-        };
-
-        public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
-        {
-            return JsonSerializer.Deserialize<T>(ref reader, options);
+            return JsonSerializer.Deserialize<Exception>(ref reader, options);
         }
 
-        public override void Write(Utf8JsonWriter writer, T value, JsonSerializerOptions options)
+        public override void Write(Utf8JsonWriter writer, Exception value, JsonSerializerOptions options)
         {
-            JsonSerializer.Serialize(writer, value.ToString(), _serializeWithoutFieldOptions);
+            writer.WriteStartObject();
+
+            writer.WriteString("ClassName", value?.GetType()?.FullName);
+            writer.WriteString("Message", value?.Message);
+            writer.WriteString("Source", value?.Source);
+            writer.WriteString("StackTraceString", value?.StackTrace);
```

Comment:
Do you also want to log Inner exception of they are not null? Maybe check to 2-3 levels instead of logging all inner exceptions.

Reply:
Good point. The portal code isn't actually using the InnerException property, but it's better to have it. It's a little tricky to check the level, so I just added the whole.

Reply:
nice addition.